lib/src/entities/user_profile.dart

```diff
@@ -9,6 +9,15 @@ part 'user_profile.g.dart';
 /// [UserProfile] represents a valid user as mapped out from the backend
 /// In the event the backend mapping changes, this model should also change as well
 
+/// [_decodeAssistant] checks the String name of the assistant passed
+/// from the backend and returns a valid [Assistant]
+Assistant _decodeAssistant(dynamic value) {
```

Comment:
The implementation and intent are not coherent.
ideally,`_decodeAssistant` expect a json string. But the implementation expects a normal string.
https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/fromJson.html

Reply:
This has been removed based on the discussion below.